src/components/swap/SwapRoute.tsx

```diff
@@ -1,64 +1,83 @@
+}
 
-  const gasPrice =
-    // TODO(WEB-2022)
-    // Can `trade.gasUseEstimateUSD` be defined when `chainId` is not in `SUPPORTED_GAS_ESTIMATE_CHAIN_IDS`?
-    trade.gasUseEstimateUSD && chainId && SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId)
-      ? trade.gasUseEstimateUSD === 0
-        ? '<$0.01'
-        : '$' + trade.gasUseEstimateUSD.toFixed(2)
-      : undefined
+function RouteLabel({ trade }: { trade: SubmittableTrade }) {
+  return (
+    <RowBetween>
+      <ThemedText.BodySmall color="neutral2">Order Routing</ThemedText.BodySmall>
+      <RouterLabel trade={trade} color="neutral1" />
+    </RowBetween>
+  )
+}
 
+function PriceImpactRow({ trade }: { trade: ClassicTrade }) {
+  const { formatPriceImpact } = useFormatter()
   return (
+    <ThemedText.BodySmall color="neutral2">
+      <RowBetween>
+        <Trans>Price Impact</Trans>
+        <div>{formatPriceImpact(trade.priceImpact)}</div>
+      </RowBetween>
+    </ThemedText.BodySmall>
+  )
+}
+
+export function RoutingTooltip({ trade }: { trade: SubmittableTrade }) {
+  return isClassicTrade(trade) ? (
+    <Column gap="md">
+      <PriceImpactRow trade={trade} />
+      <Separator />
+      <RouteLabel trade={trade} />
+      <SwapRoute trade={trade} />
+    </Column>
+  ) : (
     <Column gap="md">
-      <RouterLabel trade={trade} color="neutral2" />
+      <RouteLabel trade={trade} />
       <Separator />
-      {syncing ? (
-        <LoadingRows>
-          <div style={{ width: '100%', height: '30px' }} />
-        </LoadingRows>
-      ) : (
-        <RoutingDiagram
-          currencyIn={trade.inputAmount.currency}
-          currencyOut={trade.outputAmount.currency}
-          routes={routes}
-        />
-      )}
-      {autoRouterSupported && (
-        <>
-          <Separator />
-          {syncing ? (
-            <LoadingRows>
-              <div style={{ width: '100%', height: '15px' }} />
-            </LoadingRows>
-          ) : (
-            <ThemedText.Caption color="neutral2">
-              {gasPrice ? <Trans>Best price route costs ~{gasPrice} in gas.</Trans> : null}{' '}
-              <Trans>
-                This route optimizes your total output by considering split routes, multiple hops, and the gas cost of
-                each step.
-              </Trans>
-            </ThemedText.Caption>
-          )}
-        </>
-      )}
+      <UniswapXDescription />
+    </Column>
+  )
+}
+
+export function SwapRoute({ trade }: { trade: ClassicTrade }) {
+  const { inputAmount, outputAmount } = trade
+  const routes = getRoutingDiagramEntries(trade)
+  const gasPrice = useGasPrice(trade)
+
+  return useAutoRouterSupported() ? (
+    <Column gap="md">
+      <RoutingDiagram routes={routes} currencyIn={inputAmount.currency} currencyOut={outputAmount.currency} />
+      <ThemedText.Caption color="neutral2">
+        {gasPrice && <Trans>Best price route costs ~{gasPrice} in gas. </Trans>}
```

Comment:
```suggestion
        {Boolean(gasPrice) && <Trans>Best price route costs ~{gasPrice} in gas. </Trans>}
```

idk if this is a number type but if so this is important